cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -28,6 +31,8 @@ pub struct GrammarAST {
     pub expectrr: Option<(usize, Span)>,
     pub parse_param: Option<(String, String)>,
     pub programs: Option<String>,
+    // Unchecked `Symbol` names, with spans pointing into the `%expect-unused` declaration.
```

Comment:
I'm not quite sure what "unchecked" means here.

Reply:
I think it should probably say "Potentially invalid", e.g. the caller must check them for validity, before using them and potentially return an `YaccGrammarError::UnknownRuleRef` or `UnknownToken`, as is done on lines 218 and 226 of https://github.com/softdevteam/grmtools/pull/348/commits/4fea2b669b97966165c5131963dba30156472d9a

Reply:
Maybe we can use the idiom from https://github.com/softdevteam/grmtools/pull/348#discussion_r962587727 of "unexpectedly unused"?

Reply:
Hopefully I didn't go overboard in clarifying this in [e81fa9d](https://github.com/softdevteam/grmtools/pull/348/commits/e81fa9dfc66d6953acdf9d96c5bcd871848074e7) I didn't have much luck trimming it down, while still getting all the intricacies of it conveyed.

Reply:
I think the 2nd and 3rd sentences are repetitions of the 1st :) So just "If a `Symbol` is listed here but unused in the grammar, the fact that it is unused will be ignored. The `Symbol`s span points to the name given within a `%expect-unused` declaration." would be fine!

Reply:
I just want to triple check: 
"A symbol listed here that is not actually present in `rules` or `tokens` should be an error.",
is actually different than `unused`, which are present in the rules/tokens but unreachable from the `start_rule()`.

e.g. 
```
%expect-unused nonexistent
%%
Start: ;
```

Reply:
Anyhow I pushed the less verbose version in https://github.com/softdevteam/grmtools/pull/348/commits/a3cea98a5cb65598ef282f9e8ebf0be3eea88f79

Reply:
> I just want to triple check:
"A symbol listed here that is not actually present in rules or tokens should be an error.",
is actually different than unused, which are present in the rules/tokens but unreachable from the start_rule().

I'm not sure I understand the distinction.

Reply:
Above `NonexistentRule` is neither a rule or a token, it produces a `UnknownRuleRef` error, and `'NonexistentToken'` an `UnknownToken`.  While both `ExistsRule`, and `ExistsToken` are unused, so the distinction I was trying to highlight was one of  `unknown` vs `unused`.

```
%expect-unused NonexistentRule ExistsRule 'NonexistentToken' 'ExistsToken' 
%token ExistsToken
%%
Start: ;
ExistsRule: ;
```

Reply:
Ah, duh, I see what you mean now. Sorry!

How about something simple like "The set of symbol names that, if unused in a grammar, will not cause a warning or error"?

Reply:
Sounds good to me, I just wanted to ensure it didn't sneak through. [3b546bf](https://github.com/softdevteam/grmtools/pull/348/commits/3b546bf3c6f754490d309ee96e325c3285fcc835)